sdks/python/apache_beam/runners/worker/sdk_worker.py

```diff
@@ -570,6 +576,23 @@ def _log_lull_in_bundle_processor(self, processor):
           step_name_log,
           stack_trace)
 
+      if self._should_log_full_thread_dump():
+        self._log_full_thread_dump()
+
+  def _should_log_full_thread_dump(self):
+    now = time.time()
+    if self._last_full_thread_dump_secs + LOG_LULL_FULL_THREAD_DUMP_S < now:
+      self._last_full_thread_dump_secs = now
+      return True
+    return False
+
+  def _log_full_thread_dump(self):
```

Comment:
Can we reuse thread_dump (from https://github.com/apache/beam/blob/master/sdks/python/apache_beam/runners/worker/worker_status.py#L37) ?

Reply:
Thanks for the pointer. Done.

Reply:
This is a nit. I think at this point, you can remove _log_full_thread_dump() and directly call thread_dump() instead.

Reply:
Thanks! I'm using _log_full_thread_dump() as a mock in the unit test. I can mock thread_dump() instead there or leave it as is. Let me know your thoughts.